.github/workflows/main.yml

```diff
@@ -0,0 +1,29 @@
+name: pytorch-nightly-benchmarking
+on:
+  schedule:
+    - cron: '0 8 * * *' # run at 8 AM UTC
+
+jobs:
+  build-docker:
```

Comment:
i'm curious about how you split build-docker and build-benchmark-docker.  What is the reason to split, and what happens in each?

also curious about what you plan to do with the docker images outside of this nightly job.  if we can reuse the same already-built docker image from multiple places, it is a win.  but if we build a new image every time we use one, i'm not sure what value docker provides?

Reply:
The reason to split is because I don't want to maintain the dependencies of pytorch/torchtext/torchvision myself (e.g. cudatoolkit, cudnn) as this part is already managed by pytorch Dockerfile.
build-benchmark-docker relies on build-docker, and installs all benchmark dependencies.

You are right that we need to rebuild docker every nightly test. But I still think it is worthwhile to use docker. First, it doesn't take much time (~10 min) comparing to the benchmark running time. Second, if we need to re-run or debug some benchmarks (say one of the models has regression), then we don't need to rebuild everything because we can just use the existing one. Also, we can control the number of history dockers (say past one month) to reproduce history performance data.

Reply:
ok. i like having dockers.  i think we should make it easy for someone to find the history of dockers and clone one so they can benefit from your work.

i wonder if you need to build the pytorch docker yourself though?  can't you just use 'FROM' keyword and reference already-built pytorch nightly docker, built by existing pytorch CI?

Reply:
See above. There is no pre-existing built pytorch nightly docker, that's why we are building our own here.